dg-core/src/main/java/org/finra/datagenerator/consumer/DataConsumer.java

```diff
@@ -223,4 +223,27 @@ private String getResponse(String path) {
         }
         return content.toString();
     }
+
+    /**
+     * Consumes a produced result. Calls every transformer in sequence without
+     * calling data writers.
+     *
+     * @param initialVars a map containing the initial variables assignments
+     * @return the datapipe - number of lines written
+     */
+    public DataPipe consumeMap(Map<String, String> initialVars) {
```

Comment:
Because for default distributor we are writing data to output file. While in Spark Distributor we have to return data to distributor so we can write data to RDD,